src/diffusers/models/embeddings.py

```diff
@@ -298,14 +300,50 @@ def get_1d_rotary_pos_embed(dim: int, pos: Union[np.ndarray, int], theta: float
     freqs = 1.0 / (theta ** (torch.arange(0, dim, 2)[: (dim // 2)].float() / dim))  # [D/2]
     t = torch.from_numpy(pos).to(freqs.device)  # type: ignore  # [S]
     freqs = torch.outer(t, freqs).float()  # type: ignore   # [S, D/2]
-    if use_real:
+    if use_real and repeat_interleave_real:
         freqs_cos = freqs.cos().repeat_interleave(2, dim=1)  # [S, D]
         freqs_sin = freqs.sin().repeat_interleave(2, dim=1)  # [S, D]
         return freqs_cos, freqs_sin
+    elif use_real:
+        freqs_cos = torch.cat([freqs.cos(), freqs.cos()], dim = -1) # [S, D]
+        freqs_sin = torch.cat([freqs.sin(), freqs.sin()], dim = -1) # [S, D]
+        return freqs_cos, freqs_sin 
     else:
         freqs_cis = torch.polar(torch.ones_like(freqs), freqs)  # complex64     # [S, D/2]
         return freqs_cis
 
+def apply_partial_rotary_emb(
```

Comment:
we can just use `apply_rotary_emb` to support this

```python
def apply_rotary_emb():
    ...
    rot_dim = cos.shape[-1]
    x_unrotated = None
    if x.shape[-1] > rot_dim:
        x, x_unrotated = ..
    ....
   if x_unrotated is not None:
      out = torch.cat((out, x_unrotated), dim = -1)
    return out
```

Reply:
There's a small nuance in both methods:

1. `apply_rotary_emb` creates x_real and x_imag by reshaping like this: `x.reshape(*x.shape[:-1], -1, 2).unbind(-1)`
2.  `apply_partial_rotary_emb` reshapes in a different way (last two dimensions are swapped) : `x_to_rotate.reshape(*x_to_rotate.shape[:-1], 2, -1).unbind(dim=-2)`

The resulting tensors are totally different. Happy to solve this a different way , I could add a boolean to reshape one way or another, WDYT ?